src/script/descriptor.cpp

```diff
@@ -1093,13 +1093,37 @@ class TRDescriptor final : public DescriptorImpl
     std::optional<int64_t> ScriptSize() const override { return 1 + 1 + 32; }
 
     std::optional<int64_t> MaxSatisfactionWeight(bool) const override {
-        // FIXME: We assume keypath spend, which can lead to very large underestimations.
-        return 1 + 65;
+        // Start from the size of a keypath spend.
+        int64_t max_weight = 1 + 65;
+
+        // Then go through all the existing leaves to check if there is anything more expensive.
+        const bool dummy_max_sig = true;
+        for (size_t i = 0; i < m_subdescriptor_args.size(); ++i) {
+            // Anything inside a Tapscript leave must have its satisfaction and script size set.
+            const auto sat_size = m_subdescriptor_args[i]->MaxSatSize(dummy_max_sig);
+            const auto script_size = m_subdescriptor_args[i]->ScriptSize();
+            const auto control_size = 33 + 32 * m_depths[i];
+            const auto total_weight = 1 + control_size + 1 + *script_size + 1 + *sat_size;
```

Comment:
I assume the 1s here are the CompactSize length indicators for the witness items. It seems to me that any of the three could be more than 252 bytes, so perhaps you need to check for that and use 3 bytes for the CompactSize in case the corresponding element is longer?

Reply:
One of those 1s should also be for the entire stack size, which could also be larger than 252.

Reply:
Done.